cmd/ltkeycloak/from_mattermost.go

```diff
@@ -0,0 +1,286 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+
+	gocloak "github.com/Nerzal/gocloak/v13"
+	"github.com/mattermost/mattermost-load-test-ng/deployment"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/spf13/cobra"
+)
+
+func userTxtWriter(usersTxtChan chan string, doneChan chan struct{}, userPassword string) {
+	usersTxtFile, err := os.OpenFile("users.txt", os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		mlog.Error("failed to open users.txt file")
+	}
+	defer usersTxtFile.Close()
+
+	for {
+		select {
+		case email := <-usersTxtChan:
+			_, err := usersTxtFile.Write([]byte(model.UserAuthServiceSaml + ":" + email + " " + userPassword + "\n"))
+			if err != nil {
+				mlog.Error("failed to write to users.txt", mlog.String("err", err.Error()))
+			}
+
+		case <-doneChan:
+			return
+		}
+	}
+}
+
+type workerConfig struct {
+	workerNumber     int
+	usersChan        chan *model.User
+	wg               *sync.WaitGroup
+	mmClient         *model.Client4
+	keycloakClient   *gocloak.GoCloak
+	keycloakToken    *gocloak.JWT
+	keycloakRealm    string
+	userPassword     string
+	deploymentConfig *deployment.Config
+	doneChan         chan struct{}
+}
+
+func migrateMattermostUsersToKeycloak(worker *workerConfig) {
+	refreshTokenTicker := time.NewTicker(30 * time.Second)
+	defer refreshTokenTicker.Stop()
+
+	for {
+		select {
+		case user := <-worker.usersChan:
+			ctx, cancel := context.WithTimeout(context.Background(), operationTimeout)
+
+			kcUserID, err := worker.keycloakClient.CreateUser(ctx, worker.keycloakToken.AccessToken, worker.keycloakRealm, gocloak.User{
+				Username:      &user.Username,
+				Email:         &user.Email,
+				Enabled:       gocloak.BoolP(true),
+				EmailVerified: &user.EmailVerified,
+				Credentials: &[]gocloak.CredentialRepresentation{
+					{
+						Temporary: gocloak.BoolP(false),
+						Type:      gocloak.StringP("password"),
+						Value:     gocloak.StringP(worker.userPassword),
+					},
+				},
+			})
+			if err != nil {
+				cancel()
+				// Ignore already existing users
+				if apiErr, ok := err.(*gocloak.APIError); ok && apiErr.Code == 409 {
+					mlog.Debug("user already exists in keycloak", mlog.String("username", user.Username))
+					continue
+				}
+
+				mlog.Error("failed to create user in keycloak", mlog.String("err", err.Error()))
+				continue
+			}
+
+			user.AuthData = model.NewString(kcUserID)
```

Comment:
We can use `NewPointer` now!
```suggestion
			user.AuthData = model.NewPointer(kcUserID)
```